src/core/lib/iomgr/tcp_client_posix.cc

```diff
@@ -86,6 +100,41 @@ void do_tcp_client_global_init(void) {
   g_connection_shards = new std::vector<struct ConnectionShard>(num_shards);
 }
 
+int64_t event_engine_tcp_client_connect(
```

Comment:
This appears to be implementation-agnostic. Can it be moved out of the posix-specific code paths? Otherwise we'll _duplicate_ it for every engine implementation we want to try with shims.

Reply:
Moved it to event_engine_shims/tcp_client{.h,.cc}